plutus-core/executables/plc/Main.hs

```diff
@@ -161,16 +175,20 @@ runApplyToData (ApplyOptions inputfiles ifmt outp ofmt mode) = do
 ---------------- Typechecking ----------------
 
 runTypecheck :: TypecheckOptions -> IO ()
-runTypecheck (TypecheckOptions inp fmt) = do
+runTypecheck (TypecheckOptions inp fmt printMode printModeDeBruijn) = do
   prog <- readProgram fmt inp
   case PLC.runQuoteT $ do
     tcConfig <- PLC.getDefTypeCheckConfig ()
     PLC.inferTypeOfProgram tcConfig (void prog)
     of
       Left (e :: PLC.Error PLC.DefaultUni PLC.DefaultFun ()) ->
         errorWithoutStackTrace $ PP.displayPlc e
-      Right ty                                               ->
-        T.putStrLn (PP.displayPlc ty) >> exitSuccess
+      Right (PLC.Normalized ty)                                  ->
+        if printModeDeBruijn
+        then
+          let w = toDeBruijnTypePLC ty
+          in print (getPrintMethod printMode w) >> exitSuccess
+        else print (getPrintMethod printMode ty) >> exitSuccess
```

Comment:
Does something like this make sense? Makes it a bit cleaner imo, but feel free to ignore if you don't agree/it doesn't work.

```suggestion
let transformNames = if printModeDeBruijn then toDeBruijnTypePLC else id
in print (getPrintMethod printMode (transformNames ty)) >> exitSuccess
```

Reply:
Unfortunately that doesn't work because `ty` and `toDeBruijnTy ty` have different types (the  type of PLC types is parametric over the type of names, and `toDebruijnType` changes that).  I think we'd have to introduce a new type class to get round that.

Reply:
Looks great, thank you!